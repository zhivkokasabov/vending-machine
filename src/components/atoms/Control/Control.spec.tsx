import { fireEvent, render } from "@testing-library/react";
import { Control } from ".";

describe("Control tests", () => {
  const onClickMock = jest.fn();

  const renderComponent = () => render(
    <Control onClick={onClickMock} className="className" variant="default">
      1
    </Control>
  );

  it("should render", () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should propagate click", () => {
    const { getByRole } = renderComponent();

    fireEvent.click(getByRole("button"));

    expect(onClickMock).toHaveBeenCalled();
  });
});
