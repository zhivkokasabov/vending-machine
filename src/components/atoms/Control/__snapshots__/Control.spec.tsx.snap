// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control tests should render 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 bg-neutral-900 text-neutral-50 shadow hover:bg-neutral-900/90 dark:bg-neutral-50 dark:text-neutral-900 dark:hover:bg-neutral-50/90 px-4 py-2 className rounded-full w-[3rem] h-[3rem]"
  >
    1
  </button>
</DocumentFragment>
`;
